src/main/java/Application.java

```diff
@@ -1,21 +1,119 @@
+                        String[] deadlineParams = userInput[1].split(" /");
+                        newTask = new Event(deadlineParams[0], deadlineParams[1]);
+                        break;
+                    }
+                    }
+                    taskManager.addTask(newTask);
+                    System.out.println(UserInterface.line + "\tOne more task added to the hustle:\n\t\t" + newTask + "\n" + "\tYou now have " + taskManager.getNumOfTasks() + " tasks in total.\n" + UserInterface.line);
+                }
+
+            } catch (ToDoBeastException e) {
+                System.out.println(UserInterface.line + "\t" + e.getMessage() + "\n" + UserInterface.line);
+            } finally {
+                userInput = sc.nextLine().split(" ", 2);
+                command = userInput[0];
+            }
+        }
+        UserInterface.exitDisplay();
+    }
+
+    public void checkUserInput(String[] userInput) throws ToDoBeastException {
+        String command = userInput[0];
+        if ((command.equals("todo") || command.equals("deadline") || command.equals("event")) && userInput.length == 1) {
+            throw new ToDoBeastException("OOPS :( !!! The description of a " + command + " cannot be empty.");
+        }
+        else if (command.equals("deadline") || command.equals("event")) {
+            String[] testParams = userInput[1].split(" /");
+            if (testParams.length == 1) {
+                throw new ToDoBeastException("OOPS :( !!! Invalid format for time/date input.");
+            } else {
+                if (!testParams[1].startsWith("by") && command.equals("deadline")) {
+                    throw new ToDoBeastException("OOPS :( !!! Time/date input for deadlines should start with /by.");
+                } else if (!testParams[1].startsWith("at") && command.equals("event")) {
+                    throw new ToDoBeastException("OOPS :( !!! Time/date input for events should start with /at.");
+                }
+            }
+        }
+        else if (command.equals("done") || command.equals("delete")) {
+            if (userInput.length == 1) {
+                throw new ToDoBeastException("OOPS :( !!! The " + command + " command requires the index of the task to be processed.");
+            }
+            try {
+                int taskNumber = Integer.parseInt(userInput[1]);
+                if (taskNumber > taskManager.getNumOfTasks() || taskNumber <= 0) {
+                    throw new ToDoBeastException("OOPS :( !!! Task index provided is not present in the list.");
+                }
+            } catch (NumberFormatException e) {
+                throw new ToDoBeastException("OOPS :( !!! Task index provided is not a number.");
+            }
+        }
+        else if (!isValidCommand(command)) {
+            throw new ToDoBeastException("OOPS :( !!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public boolean isValidCommand(String command) {
```

Comment:
Maybe can rename to checkIfValidCommand? So as to make the method more verb-like?